src/kaitai_spec/attribute/tests.rs

```diff
@@ -0,0 +1,191 @@
+#[cfg(test)]
+mod tests {
+    use super::*;
+#[test]
+    fn simple_de() {
+        let yaml = "
+            id: magic1
+            doc: example
+            doc-ref: ref1
+            contents: [0xca, 0xfe, 0xba, 0xbe]
+        ";
+
+        let attribute: Attribute = serde_yaml::from_str(yaml).unwrap();
+
+        let Attribute {
+            id,
+            doc,
+            doc_ref,
+            type_,
+        } = attribute;
+
+        assert_eq!(id, "magic1");
+        assert_eq!(doc, Some("example".to_string()));
+        assert_eq!(doc_ref, Some("ref1".to_string()));
+        match type_ {
+            AttributeType::Contents(contents) => {
+                assert_eq!(contents, vec![0xCA, 0xFE, 0xBA, 0xBE]);
+            }
+        }
```

Comment:
I suggest to refactor it that way since we definitely will extend the enumeration. It would be nice to avoid any future changes in tests.
```suggestion
        let contents = match type_ {
            AttributeType::Contents(contents) => contents,
            _ => unreachable!(),
        };
        
        assert_eq!(contents, vec![0xCA, 0xFE, 0xBA, 0xBE]);
```